backend/src/main/java/gov/cdc/usds/simplereport/validators/TestResultFileValidator.java

```diff
@@ -0,0 +1,463 @@
+      errors.addAll(employedInHealthcare.getPossibleError());
+      errors.addAll(symptomaticForDisease.getPossibleError());
+      errors.addAll(illnessOnsetDate.getPossibleError());
+      errors.addAll(residentCongregateSetting.getPossibleError());
+      errors.addAll(residenceType.getPossibleError());
+      errors.addAll(hospitalized.getPossibleError());
+      errors.addAll(icu.getPossibleError());
+      errors.addAll(orderingFacilityName.getPossibleError());
+      errors.addAll(orderingFacilityStreet.getPossibleError());
+      errors.addAll(orderingFacilityStreet2.getPossibleError());
+      errors.addAll(orderingFacilityCity.getPossibleError());
+      errors.addAll(orderingFacilityState.getPossibleError());
+      errors.addAll(orderingFacilityZipCode.getPossibleError());
+      errors.addAll(orderingFacilityPhoneNumber.getPossibleError());
+      errors.addAll(comment.getPossibleError());
+      errors.addAll(testResultStatus.getPossibleError());
+
+      // validate individual values
+      errors.addAll(validateState(patientState));
+      errors.addAll(validateState(orderingProviderState));
+      errors.addAll(validateState(testingLabState));
+      errors.addAll(validateState(orderingFacilityState));
+
+      errors.addAll(validateZipCode(patientZipCode));
+      errors.addAll(validateZipCode(orderingProviderZipCode));
+      errors.addAll(validateZipCode(testingLabZipCode));
+      errors.addAll(validateZipCode(orderingFacilityZipCode));
+
+      errors.addAll(validatePhoneNumber(patientPhoneNumber));
+      errors.addAll(validatePhoneNumber(orderingProviderPhoneNumber));
+      errors.addAll(validatePhoneNumber(testingLabPhoneNumber));
+      errors.addAll(validatePhoneNumber(orderingFacilityPhoneNumber));
+
+      errors.addAll(validateDate(patientDob));
+      errors.addAll(validateDate(illnessOnsetDate));
+
+      errors.addAll(validateDateTime(orderTestDate));
+      errors.addAll(validateDateTime(specimenCollectionDate));
+      errors.addAll(validateDateTime(testingLabSpecimenReceivedDate));
+      errors.addAll(validateDateTime(testResultDate));
+      errors.addAll(validateDateTime(dateResultReleased));
+
+      errors.addAll(validateEmail(patientEmail));
+      errors.addAll(validateRace(patientRace));
+      errors.addAll(validateGender(patientGender));
+      errors.addAll(validateEthnicity(patientEthnicity));
+
+      errors.addAll(validateYesNoAnswer(pregnant));
+      errors.addAll(validateYesNoAnswer(employedInHealthcare));
+      errors.addAll(validateYesNoAnswer(symptomaticForDisease));
+      errors.addAll(validateYesNoAnswer(residentCongregateSetting));
+      errors.addAll(validateYesNoAnswer(hospitalized));
+      errors.addAll(validateYesNoAnswer(icu));
+      errors.addAll(validateResidence(residenceType));
+
+      errors.addAll(validateTestResult(testResult));
+      errors.addAll(validateTestResultStatus(testResultStatus));
+      errors.addAll(validateSpecimenType(specimenType));
+    }
```

Comment:
can we add validating CLIA? SR UI I think does a basic check that it's 11 characters and the 3rd is a D

also checking that [the comment doesn't contain any commas](https://demo.simplereport.gov/app/results/upload/submit/guide?facility=a8740505-f8c0-47e4-bd1b-e21d5dea3b4a#doc-comment)

Reply:
We can, the docs didn't specify a format and I didn't wanna guess, we can do some research to see if that's always the case